enhancements/ingress/bind-options.md

```diff
@@ -0,0 +1,268 @@
+	// be occupied by other processess
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=30000
+	// +kubebuilder:default:=1936
+	// +optional
+	StatsPort int32 `json:"statsPort,omitempty"`
+}
+```
+
+The following example configures two IngressControllers with HostNetwork strategy that can run on the same nodes without port conflicts:
+
+```yaml
+---
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default-1
+  namespace: openshift-ingress-operator
+spec:
+  endpointPublishingStrategy:
+    type: HostNetwork
+---
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default-2
+  namespace: openshift-ingress-operator
+spec:
+  endpointPublishingStrategy:
+    type: HostNetwork
+    hostNetwork:
+      bindOptions:
+        httpPort: 11080
+        httpsPort: 11443
+        sniPort: 11444
+        noSniPort: 11445
+        statsPort: 1937
+```
+
+### Validation
+
+Specifying `spec.endpointPublishingStrategy.type: HostNetwork` and omitting
+`spec.endpointPublishingStrategy.hostNetwork` or
+`spec.endpointPublishingStrategy.hostNetwork.bindOptions` is valid and specifies
+the default behavior, which is to use the default ports to bind to. The API validates
+that any value provided for the ports defined in
+`spec.endpointPublishingStrategy.hostNetwork.bindOptions` is greater than `1`, and lower than `30000` (to avoid conflicts with 30000-32767 nodePort services range).
+
+
+### User Stories
+
+#### As a cluster administrator, I need to configure bind ports for my IngressController
+
+To satisfy this use-case, the cluster administrator can set the
+IngressController's `spec.endpointPublishingStrategy.hostNetwork.bindOptions`.
+
+
+### Implementation Details
+
+Implementing this enhancement requires changes in the following repositories:
+
+* openshift/api
+* openshift/cluster-ingress-operator
+
+OpenShift Cluster Ingress Operator, creates a deployment for Router with environment variables for port bindings which Openshift Router already respects. `ROUTER_SERVICE_HTTP_PORT`, `ROUTER_SERVICE_HTTPS_PORT`, `ROUTER_SERVICE_SNI_PORT`, `ROUTER_SERVICE_NO_SNI_PORT`, `STATS_PORT`.
+
+### Risks and Mitigations
+
+
+Users might define conflicting ports which causes haproxy process fail to start. A mitigation to this risk is to implement a validation feature in cluster-ingress-operator in reconsiler loop to ensure all the ports defined in the `bindOptions` section are unique, and return err if there are conflicting ports with a meaningful error message.
+
+Also if the underlying IngressController implementation were to change away from HAProxy to a different implementation, some values such as `sniPort` and `noSniPort` might not be needed and loose their meaning. Also the `statsPort` teminology might be different for other tools such as `Envoy`, which `metricPort` might be more meaningful in that case.
```

Comment:
I agree about `sni` and `nosni` names which are pure HAProxy things but `stats` was already chosen to be a generic name detached from any concrete implementation.
BTW, it looks like Envoy is also favoring the name statistics: [statistics architecture](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/observability/statistics#arch-overview-statistics).

Reply:
the last sentence removed